### PR TITLE
feat(Pagination): [FX-2059] update pagination to the latest design

### PR DESCRIPTION
**Pull request by teimurjan**
[FX-2059](https://toptal-core.atlassian.net/browse/FX-2059)

### Description

- Added `variant` and `siblingCount` props.
  - Overriding button color [was discussed with the BASE team](https://toptal-core.slack.com/archives/CCC3GP6CC/p1630916998026800).
- Decrease spacing.
- Changed active page button color.

### How to test

1. Visit the temploy.
2. Check `Pagination` docs.
3. Compare to the designs: https://app.abstract.com/projects/1b06c884-06af-482a-bf12-a82f521a19a1/branches/master/commits/312f8be305c3d8c6c778ecb7bce0302eab1f03e5/files/e163c97f-163c-483d-904d-a4d279b01566/layers/0344789B-B331-4E30-8B06-03989D1DF742?collectionId=a80f780a-4a13-4052-bae9-d920ced900a9&collectionLayerId=090b6339-904f-4cd9-b77e-7aff15f93638&mode=build&selected=4224037875-46FC1419-DAD0-43DE-BD27-373E348392EB

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
|Screen Shot 2021-09-06 at 15 44 33|Screen Shot 2021-09-06 at 15 46 26|

### Review

- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/api-principles.md)
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure that tests pass by running `yarn test`
- [x] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

### CODE REVIEW

File: packages/picasso/src/Pagination/__snapshots__/test.tsx.snap [hunk 1]
```diff
@@ -241,7 +291,61 @@ exports[`Pagination renders nothing for 0 pages 1`] = `
 <div>
   <div
     class="Picasso-root"
-  />
```
Comment (TomasSlama) [inline packages/picasso/src/Pagination/__snapshots__/test.tsx.snap:244]: This seems as unexpected behavior to me. As tests says, it should render nothing for 0 pages. But it works only for 1 page.

File: packages/picasso/src/Pagination/story/index.jsx [hunk 2]
```diff
@@ -18,7 +18,7 @@ page.createTabChapter('Props').addComponentDocs({
     onPageChange: {
       type: {
         name: 'function',
-        description: '(pageNumber: number) => void'
+        description: '(page‰: number) => void'
```
Comment (TomasSlama) [inline packages/picasso/src/Pagination/story/index.jsx:21]: is `‰` intentional? or typo?

**Review by TomasSlama — CHANGES_REQUESTED** [review]
Looks good, only one small issue - it doesn't work correctly for 0 pages.

File: packages/picasso/src/Pagination/story/index.jsx (hunk 2 above)
Comment (teimurjan) [reply packages/picasso/src/Pagination/story/index.jsx:21]: a typo. I will fix, thanks!

**Review by TomasSlama — APPROVED** [review]

File: packages/picasso/src/PaginationButton/styles.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+import { createVariant } from '../Button'
+
+export default (theme: Theme) =>
+  createStyles({
+    paginationButton: {
+      minWidth: '1.5em',
+      padding: '0 0.3em',
+
+      '&$active': theme.palette.grey.dark
```
Comment (havenchyk) [inline packages/picasso/src/PaginationButton/styles.ts:11]: that's new to me. how is it possible `palette.grey.dark` doesn't exist?

File: packages/picasso/src/Button/index.ts [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import { Props } from './Button'
 
+export { createVariant } from './styles'
```
Comment (havenchyk) [inline packages/picasso/src/Button/index.ts:3]: this is needed for internal usage only but the user will be able to import it from '@toptal/picasso/Button', so idk if this export should be here or not

File: packages/picasso/src/Pagination/utils/get-siblings/test.ts [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import getSiblings from './get-siblings'
+
+describe('getSiblings', () => {
+  it('gets first-level siblings', () => {
+    expect(
+      getSiblings({ activePage: 1, totalPages: 5, siblingCount: 1 })
+    ).toStrictEqual([[], [2]])
```
Comment (havenchyk) [inline packages/picasso/src/Pagination/utils/get-siblings/test.ts:7]: is `toStrictEqual` a preferable way now? I see no difference with `toEqual` for our cases

File: packages/picasso/src/PaginationButton/PaginationButton.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+  const {
+    paginationButton: paginationButtonClass,
+    active: activeClass
+  } = useStyles()
+
+  return (
+    <Button
+      classes={{
+        root: cx(paginationButtonClass, classes?.root, {
```
Comment (havenchyk) [inline packages/picasso/src/PaginationButton/PaginationButton.tsx:31]: can you explain, why `classes?.root` are here? seems useless to me

**Review by havenchyk — APPROVED** [review]
LGTM, just a couple of questions/comments

**Comment by havenchyk** [comment]
@teimurjan oh, wait, we collect breaking changes in the PR https://github.com/toptal/picasso/pull/2157
please rebase on top of `feature/picasso-compliance` branch

File: packages/picasso/src/PaginationButton/styles.ts (hunk 3 above)
Comment (teimurjan) [reply packages/picasso/src/PaginationButton/styles.ts:11]: Grey in Material UI is optional, that's why it potentially can be undefined.

File: packages/picasso/src/PaginationButton/PaginationButton.tsx (hunk 6 above)
Comment (teimurjan) [reply packages/picasso/src/PaginationButton/PaginationButton.tsx:31]: `Pagination` overrides `Button`'s margin. That's why it accepts `classes`.

https://github.com/toptal/picasso/blob/9a36a1ab9b3eb83885900d53d6adfaed63a892b4/packages/picasso/src/Pagination/styles.ts#L5

File: packages/picasso/src/PaginationButton/styles.ts (hunk 3 above)
Comment (havenchyk) [reply packages/picasso/src/PaginationButton/styles.ts:11]: that's true, I checked briefly and didn't find a way to override this declaration from MUI. But I found that in our code base we used `theme.palette.grey.dark!` to bypass it, I think here it also can be used

File: packages/picasso/src/PaginationButton/PaginationButton.tsx (hunk 6 above)
Comment (havenchyk) [reply packages/picasso/src/PaginationButton/PaginationButton.tsx:31]: you're right, but from where do you expect `classes.root` will come? useStyles are created from styles of PaginationButton, not Button

**Review by havenchyk — APPROVED** [review]
Great job!

**Comment by TomasSlama** [comment]
@toptal-bot run all

**Comment by teimurjan** [comment]
@toptal-bot run visual

**Pull request merged**